functionsChallenges.js

```diff
@@ -0,0 +1,74 @@
+function maxOrMin(max, x, y) {
+    if (max && x > y) {
+        return x
+    } else if (max && y > x) {
+        return y
+    }
+}
+alert(maxOrMin(true,9,7))
+
+
+
+
+
+
+var arr = [1,4,6,8,9,7]
+function sillySum (array) {
+	var newArr =[]; 
+	for (var i = 0; i < array.length; i++) {
+		
+		 newArr.push(i*array[i]); 
+
+	}
+	return newArr }
+
+sillySum(arr)
+[0, 4, 12, 24, 36, 35]
+
+
+
+
+
+
+
+var arr = [1,4,6,8,9,7]
+function numSquare (array) {
+	var newArr =[]; 
+	for (var i = 0; i < array.length; i++) {
+		
+		 newArr.push(array[i++]*array[i++]); 
+
+	}
+	return newArr }
+
+
+
+
+
+
+
+	function isPrime (num) {
+	 
+		if (num%num==0) && (num%1==0) {
+            console.log("is prime");}
+    else {
+    	console.log("is not prime");
+    }
+
+	
+
+
+
+
+
+
+
+function primesMax (array) {
```

Comment:
We want this function to take in one number `max` and return an array of primes up until that max. Something like this:

```js
primes(100);
//=> [2, 3, 5, 7]
```

Try this as a base:

```js
function primes(max) {
  // define a variable called `primes` that's initially an empty array
  // write a `for` loop that goes from 0 up until `max`
  // inside the loop, pass each number into your `isPrime` function (which you're already doing!)
  // if the number is a prime, push it into the `primes` array
  // when you're loop is finished, return the `primes` array
}
```